### PR TITLE
[Ray] Support new ray cluster through ray client

### PR DESCRIPTION
**Pull request by fyrestone**
## What do these changes do?
Support new ray cluster through ray client.

## Related issue number

Fixes https://github.com/mars-project/mars/issues/2977

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

**Comment by qinxuye** [comment]
@chaokunyang could you please review this PR?

File: mars/oscar/backends/ray/communication.py [hunk 1]
```diff
@@ -273,9 +280,15 @@ async def handle_task(message: Any, object_ref: "ray.ObjectRef"):
                 try:
                     result = await object_ref
                 except Exception as e:  # pragma: no cover
+                    # The error ClientObjectRef can't be formatted, so
+                    # we give it a string `ClientObjectRef` instead.
+                    try:
+                        object_ref_str = str(object_ref)
+                    except Exception:
+                        object_ref_str = "ClientObjectRef"
```
Comment (chaokunyang) [inline mars/oscar/backends/ray/communication.py:288]:
```suggestion
                        object_ref_str = repr(object_ref)
```

Comment (fyrestone) [reply mars/oscar/backends/ray/communication.py:288]: Any operations on the object_ref will raise an exception, this is the fallback path, we'd better to use a plain text here.

**Review by chaokunyang — APPROVED** [review]
LGTM

File: mars/deploy/oscar/tests/test_ray.py [hunk 2]
```diff
@@ -236,11 +240,55 @@ def new_ray_session_test():
...
+        proc = subprocess.Popen([sys.executable, "-u", f.name], stdout=subprocess.PIPE)
+
+        try:
+
+            def _check_ready(expect_exit=False):
+                while True:
+                    line = proc.stdout.readline()
+                    if proc.returncode is not None:
+                        if expect_exit:
+                            break
```
Comment (chaokunyang) [inline mars/deploy/oscar/tests/test_ray.py:267]: Seems `expect_exit` not defined?

**Review by qinxuye — APPROVED** [review]
LGTM

**Pull request merged**